tests/NLog.UnitTests/NLogTraceListenerTests.cs

```diff
@@ -287,6 +287,49 @@ public void ForceLogLevelTest()
             AssertDebugLastMessage("debug", "MySource1 Warn Mary had a little lamb 0");
         }
 
+        [Fact]
+        public void FilterTrace()
+        {
+            LogManager.Configuration = CreateConfigurationFromString(@"
+                <nlog>
+                    <targets><target name='debug' type='Debug' layout='${logger} ${level} ${message} ${event-context:EventID}' /></targets>
+                    <rules>
+                        <logger name='*' minlevel='Trace' writeTo='debug' />
+                    </rules>
+                </nlog>");
+
+            TraceSource ts = CreateTraceSource();
+            Func<TraceEventCache, string, TraceEventType, int, string, object[], object, object[], bool> resultFunc =
+                (cache, source, eventType, id, formatOrMessage, args, data1, data) =>
```

Comment:
I have to agree that this func has a lot of arguments! As general rule, 7 is the max. Can this be improved?